scipy/stats/_continuous_distns.py

```diff
@@ -7964,6 +7965,136 @@ def _sf(self, x, chi):
         """
         return _argus_phi(chi * np.sqrt(1 - x**2)) / _argus_phi(chi)
 
+    def _rvs(self, chi):
+        chi = np.asarray(chi)
+        if chi.size == 1:
+            if chi <= 26:
+                out = self._rvs_scalar_small_chi(chi, self._size)
+            else:
+                out = self._rvs_scalar_large_chi(chi, self._size)
+        else:
+            shp, bc = _check_shape(chi.shape, self._size)
+            numsamples = int(np.prod(shp))
+            out = np.empty(self._size)
+            it = np.nditer([chi],
+                           flags=['multi_index'],
+                           op_flags=[['readonly']])
+            while not it.finished:
+                idx = tuple((it.multi_index[j] if not bc[j] else slice(None))
+                            for j in range(-len(self._size), 0))
+                if it[0] <= 26:
+                    r = self._rvs_scalar_small_chi(it[0], numsamples)
+                    out[idx] = r.reshape(shp)
+                else:
+                    r = self._rvs_scalar_large_chi(it[0], numsamples)
+                    out[idx] = r.reshape(shp)
+                it.iternext()
+
+        if self._size == ():
+            out = out[()]
+        return out
+
+    def _rvs_scalar_small_chi(self, chi, numsamples=None):
+        # use Vaduva's rejection method, see Devroye:
+        # Non-Uniform Random Variate Generation, 1986, section II.3.2.
+        # write: self.pdf = c * g(x) * (1 - H(x)), where
+        # H is a distribution function and g is a density
+        # g(x) = d1 * chi**2 * x * exp(-chi**2 * (1 - x**2) / 2), 0 <= x <= 1
+        # H(x) = 1 - sqrt(1 - x**2), 0 <= x <= 1
+        # Integrating g, we get:
+        # G(x) = d1 * exp(-chi**2 * (1 - x**2) / 2) - d2
+        # d1 and d2 are determined by G(0) = 0 and G(1) = 1
+        # d1 = 1 / (1 - exp(-0.5 * chi**2))
+        # d2 = 1 / (exp(0.5 * chi**2) - 1)
+        # => G(x) = (exp(chi**2 * x**2 /2) - 1) / (exp(chi**2 / 2) - 1)
+        # expected number of iterations is c
+        # c = -np.expm1(-0.5 * chi**2) * chi / (_norm_pdf_C * _argus_phi(chi))
+        # note that both G and H can be inverted easily, so we can sample
+        # rvs from these distributions
+        # H_inv(y) = sqrt(1 - (1 - y)**2)
+        # G_inv(y) = sqrt(2 * log(1 + A * y) / chi**2), A = exp(c**2 / 2) - 1
```

Comment:
The comments states `A = exp(c**2 / 2) - 1` but the code constructs `A = np.expm1(chi**2 / 2)`.  
Is `c` in the comment meant to be `chi`? Or the `chi` in the code meant to be `c`?